### PR TITLE
[ENH]: Add mrdegibbs and dwibiascorrect from mrtrix3

### PR DESCRIPTION
**Pull request by josephmje**
## Summary
Addition of dwibiascorrect references  #1126

## List of changes proposed in this PR (pull-request)
- output noise map from dwidenoise
- add classes for mrdegibbs and dwibiascorrect
- remove '.nii.gz' from eddy quad folder

## Acknowledgment
I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: nipype/interfaces/mrtrix3/preprocess.py [hunk 1]
```diff
@@ -75,6 +80,163 @@ class DWIDenoise(MRTrix3Base):
...
+    For more information, see
+    <https://mrtrix.readthedocs.io/en/latest/reference/commands/mrdegibbs.html>
+
+    Example
+    -------
+
+    >>> import nipype.interfaces.mrtrix3 as mrt
+    >>> unring = mrt.MRDeGibbs()
+    >>> unring.inputs.in_file = 'dwi.mif'
+    >>> unring.cmdline                               # doctest: +ELLIPSIS
```
Comment (mgxd) [inline nipype/interfaces/mrtrix3/preprocess.py]: i don't think you need the `ELLIPSIS` option

File: nipype/interfaces/mrtrix3/preprocess.py [hunk 2]
```diff
@@ -26,18 +26,23 @@ class DWIDenoiseInputSpec(MRTrix3BaseInputSpec):
...
-        argstr="%s",
+        argstr='%s',
         position=-1,
-        desc="the output denoised DWI image")
+        desc='the output denoised DWI image',
+        genfile=True)
 
 class DWIDenoiseOutputSpec(TraitedSpec):
-    out_file = File(desc="the output denoised DWI image", exists=True)
+    out_noisemap = File(desc='the output noise map', exists=True)
```
Comment (mgxd) [inline nipype/interfaces/mrtrix3/preprocess.py]: is this always output?

File: nipype/interfaces/mrtrix3/preprocess.py [hunk 3]
```diff
@@ -75,6 +80,163 @@ class DWIDenoise(MRTrix3Base):
...
+    out_file = File(name_template='%s_biascorr',
+        name_source='in_file',
+        keep_extension=True,
+        argstr='%s',
+        position=-1,
+        desc='the output bias corrected DWI image',
+        genfile=True)
+
+class DWIBiasCorrectOutputSpec(TraitedSpec):
+    out_bias = File(desc='the output estimated bias field')
```
Comment (mgxd) [inline nipype/interfaces/mrtrix3/preprocess.py]: is this always output?

File: nipype/interfaces/mrtrix3/preprocess.py [hunk 4]
```diff
@@ -75,6 +80,163 @@ class DWIDenoise(MRTrix3Base):
...
+    For more information, see
+    <https://mrtrix.readthedocs.io/en/latest/reference/scripts/dwibiascorrect.html>
+
+    Example
+    -------
+
+    >>> import nipype.interfaces.mrtrix3 as mrt
+    >>> bias_correct = mrt.DWIBiasCorrect()
+    >>> bias_correct.inputs.in_file = 'dwi.mif'
+    >>> bias_correct.cmdline                           # doctest: +ELLIPSIS
```
Comment (mgxd) [inline nipype/interfaces/mrtrix3/preprocess.py]: ellipsis not needed

**Comment by josephmje** [comment]
Thanks for your feedback @mgxd! I reverted my variable name change to dwidenoise in case it affects existing code using it. Also modified the arguments to the optional output files.

**Comment by mgxd** [comment]
tests should start passing again if you merge with master

File: nipype/interfaces/mrtrix3/preprocess.py [hunk 5]
```diff
@@ -28,16 +28,17 @@ class DWIDenoiseInputSpec(MRTrix3BaseInputSpec):
         desc='set the window size of the denoising filter. (default = 5,5,5)')
     noise = File(
```
Comment (mgxd) [inline nipype/interfaces/mrtrix3/preprocess.py]: it doesn't look like this is currently being tracked as an output by the interface, but if this is set it I think `dwidenoise` would make it. You should :
1. re-add it to the OutputSpec (`noise = File(..., exists=True`)
2. add a `_list_outputs()` method that checks if `self.inputs.noise` is defined, and if so adds it to outputs.

Here's an example of (2) - though you won't have to generate a filename.

**Comment by josephmje** [comment]
@mgxd I finished incorporating your suggestions. Let me know if there's anything else I'm missing.

**Pull request merged**